x-pack/legacy/plugins/actions/server/builtin_action_types/slack.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Joi from 'joi';
+import { IncomingWebhook } from '@slack/webhook';
+
+import { ActionType, ActionTypeExecutorOptions } from '../types';
+
+// TODO: figure out the right way type-safe way to mock slack's IncomingWebhook class
+let IncomingWebhookImpl: any = IncomingWebhook;
+
+// for testing
+export function setIncomingWebhookImpl(incomingWebHook: any = IncomingWebhook): void {
```

Comment:
apparently `jest.mock(moduleName)` isn't "global", it only applies to imports in the file with the `jest.mock()`, which seems kinda pointless to me: [reference](https://jestjs.io/docs/en/jest-object#jestmockmodulename-factory-options)

> Modules that are mocked with jest.mock are mocked only for the file that calls jest.mock. Another file that imports the module will get the original implementation even if it runs after the test file that mocks the module.

So that made mocking the slack module too unwieldy.  Of course, the existing bit of hacking in an alternative implementation was also a bit icky, so I ended up exposing a `getActionType()` method that allows the executor to be replaced, for testing purposes (uses the slack one by default).

Still not completely happy with this, but seems like the best approach so far.